src/org/labkey/test/tests/TextChoiceSampleTypeTest.java

```diff
@@ -0,0 +1,653 @@
+                expectedConvertedValues, actualValues);
+
+        actualValues = fieldRow.getLockedTextChoiceValues();
+
+        LabKeyAssert.assertEqualsSorted("All of the converted values should show as locked, they do not. Fatal error.",
+                expectedConvertedValues, actualValues);
+
+        updatePage.clickSave();
+
+        log("Updating field should have worked.");
+
+    }
+
+    final String STR_FIELD_NAME = "Str";
+
+    private TestDataGenerator createSampleType(String sampleTypeName, String sampleNamePrefix, String textChoiceFieldName, List<String> textChoiceValues)
+    {
+        log(String.format("Create a new sample type named '%s'.", sampleTypeName));
+        SampleTypeDefinition sampleTypeDefinition = new SampleTypeDefinition(sampleTypeName);
+        sampleTypeDefinition.setNameExpression(String.format("%s${genId}", sampleNamePrefix));
+
+        log(String.format("Add a TextChoice field named '%s'.", textChoiceFieldName));
+        FieldDefinition textChoiceField = new FieldDefinition(textChoiceFieldName, ColumnType.TextChoice);
+        textChoiceField.setTextChoiceValues(textChoiceValues);
+
+        sampleTypeDefinition.addField(textChoiceField);
+
+        log(String.format("Add a simple string field '%s' just because.", STR_FIELD_NAME));
+        FieldDefinition strField = new FieldDefinition(STR_FIELD_NAME, ColumnType.String);
+        sampleTypeDefinition.addField(strField);
+
+        return SampleTypeAPIHelper.createEmptySampleType(getCurrentContainerPath(), sampleTypeDefinition);
+
+    }
+
+    /**
+     * <p>
+     *     Test the update and deleting of TextChoice values.
+     * </p>
+     * <p>
+     *     This test will:
+     *     <ul>
+     *         <li>Validate assigned values are locked.</li>
+     *         <li>The delete button is disabled for locked values.</li>
+     *         <li>Delete an unlocked value.</li>
+     *         <li>Update a locked value and validate the update message</li>
+     *         <li>Validate that the updated value is shown on the samples overview page.</li>
+     *         <li>Update an unlocked value.</li>
+     *         <li>Updating a value to an existing value causes an error.</li>
+     *         <li>Updating/saving a changed sample type design does not update the TextChoice value if the 'Apply' button wasn't clicked.</li>
+     *     </ul>
+     * </p>
+     * @throws IOException Can be thrown by the create API call.
+     * @throws CommandException Can be thrown by the create API call.
+     */
+    @Test
+    public void testUpdatingAndDeletingValuesInSampleType() throws IOException, CommandException
+    {
+        final String sampleTypeName = "TC_Value_Updates";
+        final String textChoiceFieldName = "TextChoice_Field_1";
+        final String namePrefix = "TCE_";
+
+        // Some TextChoice values.
+        List<String> expectedUnLockedValues = new ArrayList<>();
+        expectedUnLockedValues.add("ÅÅ");
```

Comment:
Use `\u` syntax for special characters in java Strings.

Reply:
Yep, missed it.